src/main/java/com/aws/greengrass/telemetry/MetricsAggregator.java

```
@@ -213,11 +207,11 @@ protected Map<Long, List<AggregatedNamespaceData>> getMetricsToPublish(long last
                                 "timestamp": 1600127641506,
                                 "cause": null
                             } */
-                            GreengrassLogMessage egLog = objectMapper.readValue(log,
-                                    GreengrassLogMessage.class);
-                            AggregatedNamespaceData am = objectMapper.readValue(egLog.getMessage(),
-                                    AggregatedNamespaceData.class);
-                            // Avoid the metrics that are aggregated at/after the currTimestamp and before the
+                            GreengrassLogMessage egLog = objectMapper.readValue(log, GreengrassLogMessage.class);
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `ObjectMapper.readValue()` method can throw a `JsonProcessingException`, but this exception is not handled properly here. By simply logging the error and returning an empty value, the exception is ignored. This can mask potential JSON parsing issues, leading to unexpected application behavior or data corruption because the system may continue with invalid or incomplete data. To fix this, catch `JsonProcessingException` and either handle it appropriately or rethrow it as a custom runtime exception. [https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

Similar issue at line number 211.